testutils/source.go

```diff
@@ -0,0 +1,193 @@
+package testutils
+
+// CodeSample encapsulates a snippet of source code that compiles, and how many errors should be detected
+type CodeSample struct {
+	Code   string
+	Errors int
+}
+
+var (
```

Comment:
Do you plan to add more code samples here to cover all the rules? I am asking this because the tests of each rule were removed.

Reply:
Yes. This is something I need to finish doing.  There are also some placeholder tests that I need to get finished also (currently skipping).

Reply:
Cool!